### PR TITLE
Added navigation component + styles.

### PR DESCRIPTION
**Pull request by kefimochi**
Made Sass work w/ Gatsby by adding it inside plugins.
Switched images from a private directory to public since the blog itself is public.
This navigation is yet to be mobile first and look as expected but it 🌸 works 🌸

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,9 @@
     "gatsby-cli": "^2.7.40",
     "gatsby-image": "^2.2.8",
     "gatsby-plugin-react-helmet": "^3.1.5",
+    "gatsby-plugin-sass": "^2.1.13",
     "gatsby-source-filesystem": "^2.1.17",
+    "node-sass": "^4.12.0",
```
Comment (lynncyrin) [inline package.json]: non-blocking FYI: this will get swapped to `dart-sass` by the time the work on the website is done. I'm mentioning this for recordkeeping purposes, not suggested it be changed here 🙂

File: gatsby-config.js [hunk 2]
```diff
@@ -6,11 +6,12 @@ module.exports = {
   },
   plugins: [
     `gatsby-plugin-react-helmet`,
+    `gatsby-plugin-sass`,
     {
       resolve: `gatsby-source-filesystem`,
       options: {
         name: `images`,
-        path: `${__dirname}/src/images`,
```
Comment (lynncyrin) [inline gatsby-config.js]: ⭐⭐⭐ _(must change)_ Undo this change, I believe it's half of why CI is failing

Comment (lynncyrin) [inline gatsby-config.js]: ⭐⭐⭐ _(must change)_ Also undo the change where you remove the gatsby image file, I think that's the other half of why CI is failing.

Comment (lynncyrin) [reply gatsby-config.js]: In case that comment ^ isn't clear enough, what I mean is: keep the folder name as `src/images`. The `public/images` folder doesn't need to specifically be created yet, as far as I can tell.

Comment (kefimochi) [reply gatsby-config.js]: Done ✅

Comment (kefimochi) [reply gatsby-config.js]: Done ✅

File: gatsby-config.js [hunk 3]
```diff
@@ -1,11 +1,12 @@
 module.exports = {
   siteMetadata: {
     title: `Lynn Cyrin`,
-    description: `QTPoC \\ software engineer \\ organizer`,
```
Comment (lynncyrin) [inline gatsby-config.js]: ⭐⭐⭐ _(must change)_ It looks like you reset to the `main` branch without pulling in one of the commits https://github.com/lynncyrin/lynncyrin-dot-me/commit/911bfb58a43a8615ca880cffe1e3614b9bc0584b. So you'll want to manually undo this change probably.

File: src/pages/index.js [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
 import Layout from "../components/layout"
 import SEO from "../components/seo"
 
 const IndexPage = () => (
-  <Layout>
-    <SEO title="Home" />
+  <>
+    <Navigation />
+    {/* <Layout> */}
+    {/* <SEO title="Home" /> */}
```
Comment (lynncyrin) [inline src/pages/index.js]: ⭐⭐⭐ _(must change)_ The SEO tag is what's creating opengraph stuff for us, so we don't want to comment it out.

```suggestion
    <SEO title="Home" />
```

File: src/components/Navigation.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import React, { useState } from "react"
+import { Link } from "gatsby"
+import "../sass/Navigation.scss"
```
Comment (lynncyrin) [inline src/components/Navigation.js]: ⭐ _(nice to have)_ js generally starts filename with a lowercase letter

```suggestion
import "../sass/navigation.scss"
```

File: src/pages/index.js [hunk 6]
```diff
@@ -18,7 +20,8 @@ const IndexPage = () => (
         the old website
       </a>
     </p>
-  </Layout>
+    {/* </Layout> */}
```
Comment (lynncyrin) [inline src/pages/index.js]: ⭐⭐ _(seriously consider)_ If you're going to remove the layout code, I would dramatically prefer it be deleted (and it's file deleted) rather than commented out.

File: src/sass/Variables.scss [hunk 7]
```diff
@@ -0,0 +1,20 @@
+$light-blue: #dff0ea;
+$mid-blue: #95adbe;
+$mid-purple: #574f7d;
+$dark-purple: #4f3a65;
+
+$white: #e0e0e0;
+
+// 350px minimum for mobile first
```
Comment (lynncyrin) [inline src/sass/Variables.scss]: ⭐ _(nice to have)_ psychologically I understand why this comment is here, but I don't think we actually need to mention in the code what our target minimum width is.

File: src/sass/Navigation.scss [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+      font-size: 2em;
+    }
+  }
+
+  @include tablet {
+    height: 50%;
+  }
+
+  @include tablet {
+    // background: @mid-blue;
```
Comment (lynncyrin) [inline src/sass/Navigation.scss]: ⭐⭐ _(seriously consider)_ I'd prefer this removed instead of commented out

File: src/sass/Navigation.scss [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+  background-attachment: fixed;
+  background-size: 100%;
+}
+
+.logoName {
+  width: 15em;
+  height: 100%;
+  display: flex;
+  text-decoration: none;
+  // background: rebeccapurple;
```
Comment (lynncyrin) [inline src/sass/Navigation.scss]: ⭐⭐ _(seriously consider)_ same as elsewhere, I'd prefer this removed

File: src/sass/Navigation.scss [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+  }
+
+  h2 {
+    color: white;
+    font-size: 1.7em;
+    text-decoration: none;
+    text-align: center;
+    padding-left: 0.5em;
+    text-decoration: none;
+    font-family: "Darker Grotesque", sans-serif;
```
Comment (lynncyrin) [inline src/sass/Navigation.scss]: ⭐⭐⭐ _(must change)_ this `font-family` value (and the other one) need to go inside a variable

File: src/sass/Navigation.scss [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+nav {
+  top: 0;
+  width: 100%;
+  height: 4em;
+  display: flex;
+  align-items: center;
+  flex-direction: row;
+  justify-content: space-around;
+  position: fixed;
+  z-index: 10;
```
Comment (lynncyrin) [inline src/sass/Navigation.scss]: ⭐ _(nice to have)_ I can't tell if the `z-index` is strictly required here? If it is, can you add a comment explaining why?

File: src/components/Navigation.js [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+
+function Navigation(props) {
+  return (
+    <nav>
+      <Link to="/" activeClassName="activeNavButton" className="logoName">
+        <img
+          src="images/cool-logo.svg"
+          alt="An abstract shape with lines and circles Tech by Wenjie Jiang from the Noun Project"
+        />
+        <h2>Lynn's Blog</h2>
```
Comment (lynncyrin) [inline src/components/Navigation.js]: ⭐ _(nice to have)_ Github's JSX syntax highlighting is breaking after this line, I'm guessing because of the `'`? It'd be nice if we could fix that.

File: src/components/Navigation.js [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+import "../sass/Navigation.scss"
+
+function Navigation(props) {
+  return (
+    <nav>
+      <Link to="/" activeClassName="activeNavButton" className="logoName">
+        <img
+          src="images/cool-logo.svg"
+          alt="An abstract shape with lines and circles Tech by Wenjie Jiang from the Noun Project"
+        />
```
Comment (lynncyrin) [inline src/components/Navigation.js]: ⭐⭐⭐ _(must change)_

2 things:

- this needs a link to the source of the image _(perhaps in the alt text? in a comment? in a title? I dunno)_
- I don't believe this image was actually added in this PR?

**Comment by lynncyrin** [comment]
meta comment: I haven't seen what the changes look like visually yet, I'll do that after we go through all the ⭐️⭐️⭐️ must change implementation bits.

File: src/components/Navigation.js (hunk 13 above)
Comment (kefimochi) [reply src/components/Navigation.js]: This image was added in the public directory since you did mention previously having it, not sure if I should switch it back to the private one. If so, how would the `src` look like? When accessing images in public directory, all it takes is to declare going into images folder and derive the file from there.
You're right! I should give credit to the author(lets do alt text?). Do you think alt text by itself is decent? Any suggestions on how to make it better?

Comment (lynncyrin) [reply src/components/Navigation.js]: > If so, how would the src look like? 

```
src/images/gatsby-icon.png
src/images/cool-logo.svg
```

> You're right! I should give credit to the author(lets do alt text?). Do you think alt text by itself is decent? Any suggestions on how to make it better?

What you have here is great! 👍 👍 👍

File: gatsby-config.js (hunk 3 above)
Comment (kefimochi) [reply gatsby-config.js]: Changed it  ✅

File: src/pages/index.js (hunk 4 above)
Comment (kefimochi) [reply src/pages/index.js]: Solved  ✅

File: src/components/Navigation.js (hunk 5 above)
Comment (kefimochi) [reply src/components/Navigation.js]: Changed all file names to lowercase, both js and scss  ✅

File: src/sass/Variables.scss (hunk 7 above)
Comment (kefimochi) [reply src/sass/Variables.scss]: The target minimum is just for my reference so that I wouldn't forget about it, I'll delete it once we deploy to production or once I finish writing jsx/sass. It kind of serves as a reminder that it is mobile first blog

File: src/sass/Navigation.scss (hunk 8 above)
Comment (kefimochi) [reply src/sass/Navigation.scss]: A lot of things I out comment are there for testing purposes when building flexbox layouts  😊 It all will be cleaned up after that particular component is finished and polished, so will remove after navigation is mobile first, works on all devices and looking good!

File: src/sass/Navigation.scss (hunk 10 above)
Comment (kefimochi) [reply src/sass/Navigation.scss]: Good idea, didn't think about that! Changed  ✅Fonts are not imported yet, I'll do it on a separate branch later

File: src/sass/Navigation.scss (hunk 11 above)
Comment (kefimochi) [reply src/sass/Navigation.scss]: I always add it on navigations in case `position: absolute` will be used anywhere or in case we'll need to have something in front of other elements(this is the case for all cool slanted section edges). Not sure if it will be needed for the designs I created but I don't think it will hurt to have it as a precaution  🤷‍♀️

File: src/pages/index.js (hunk 6 above)
Comment (kefimochi) [reply src/pages/index.js]: Deleted it  ✅

File: src/sass/Navigation.scss (hunk 9 above)
Comment (kefimochi) [reply src/sass/Navigation.scss]: I'll try finishing navigation over tonight, so consider it done in 24 hours!

File: src/components/Navigation.js (hunk 13 above)
Comment (kefimochi) [reply src/components/Navigation.js]: I'll need help with sourcing the logo image, autocomplete tells me another thing but neither works  🤷‍♀️

File: src/components/Navigation.js (hunk 12 above)
Comment (kefimochi) [reply src/components/Navigation.js]: It doesn't look broken on my VSCode but I tried splitting it and hope it will work

File: src/pages/index.js [hunk 14]
```diff
@@ -18,7 +19,7 @@ const IndexPage = () => (
         the old website
       </a>
     </p>
-  </Layout>
```
Comment (lynncyrin) [inline src/pages/index.js]: ⭐⭐⭐ _(must change)_ The layout tag hasn't been completely removed, you remove the literal text `Layout` but then let in some empty tags that still need to be removed. You also need to remove the import (rather than commenting it out).

File: src/components/nav.js [hunk 15]
```diff
@@ -0,0 +1,37 @@
+import React, { useState } from "react"
+import { Link } from "gatsby"
+import "../sass/nav.scss"
+// import "../images/" --> this is what autocompleted told me
```
Comment (lynncyrin) [inline src/components/nav.js]:
```suggestion
```

⭐⭐⭐ _(must change)_ remove this

File: src/pages/404.js [hunk 16]
```diff
@@ -1,14 +1,12 @@
...
 import SEO from "../components/seo"
 
 const NotFoundPage = () => (
-  <Layout>
+  <>
     <SEO title="404: Not found" />
     <h1>NOT FOUND</h1>
     <p>You just hit a route that doesn&#39;t exist... the sadness.</p>
-  </Layout>
+  </>
```
Comment (lynncyrin) [inline src/pages/404.js]:
```suggestion
```

⭐⭐⭐ _(must change)_ remove this empty tag

File: src/pages/404.js [hunk 17]
```diff
@@ -1,14 +1,12 @@
 import React from "react"
-
-import Layout from "../components/layout"
 import SEO from "../components/seo"
 
 const NotFoundPage = () => (
-  <Layout>
+  <>
```
Comment (lynncyrin) [inline src/pages/404.js]:
```suggestion
```
```suggestion
```

⭐⭐⭐ _(must change)_ remove this empty tag too

File: src/pages/index.js [hunk 18]
```diff
@@ -1,11 +1,11 @@
 import React from "react"
 import { Link } from "gatsby"
-
-import Layout from "../components/layout"
+import Navigation from "../components/nav"
 import SEO from "../components/seo"
 
 const IndexPage = () => (
-  <Layout>
+  <>
```
Comment (lynncyrin) [inline src/pages/index.js]:
```suggestion
```

⭐⭐⭐ _(must change)_ remove this empty tag

File: src/pages/index.js [hunk 19]
```diff
@@ -18,7 +18,7 @@ const IndexPage = () => (
         the old website
       </a>
     </p>
-  </Layout>
+  </>
```
Comment (lynncyrin) [inline src/pages/index.js]:
```suggestion
```

⭐⭐⭐ _(must change)_ remove this empty tag
